framework/contrib/pp/pp.py

```diff
@@ -541,37 +536,33 @@ def get_stats(self):
                     stat.time = 0.0
         return self.__stats
 
-    def collect_stats_in_list(self):
-        """collect job execution statistics in list. Useful for benchmarking on
+    def print_stats(self):
+        """Prints job execution statistics. Useful for benchmarking on
            clusters"""
-        returnList = []
-        returnList.append("Job execution statistics:")
+
+        print "Job execution statistics:"
         walltime = time.time() - self.__creation_time
         statistics = self.get_stats().items()
         totaljobs = 0.0
         for ppserver, stat in statistics:
             totaljobs += stat.njobs
-        returnList.append(" job count | % of all jobs | job time sum | time per job | job server")
+        print " job count | % of all jobs | job time sum | " \
+                "time per job | job server"
         for ppserver, stat in statistics:
             if stat.njobs:
-                returnList.append("    %6i |        %6.2f |     %8.4f |  %11.6f | %s" \
+                print "    %6i |        %6.2f |     %8.4f |  %11.6f | %s" \
                         % (stat.njobs, 100.0*stat.njobs/totaljobs, stat.time,
-                        stat.time/stat.njobs, ppserver, ))
-        returnList.append("Time elapsed since server creation " + str(walltime))
-        returnList.append(str(self.__active_tasks)+ " active tasks," +  str(self.get_ncpus()) + " cores")
-        #print self.__active_tasks, "active tasks,", self.get_ncpus(), "cores"
+                        stat.time/stat.njobs, ppserver, )
+        print "Time elapsed since server creation", walltime
+        print self.__active_tasks, "active tasks,", self.get_ncpus(), "cores"
 
         if not self.__accurate_stats:
-            returnList.append("WARNING: statistics provided above is not accurate due to job rescheduling")
-        return returnList
-
-
-    def print_stats(self):
-        """Prints job execution statistics. Useful for benchmarking on
-           clusters"""
-        for row in self.collect_stats_in_list(): print(row)
+            print "WARNING: statistics provided above is not accurate" \
+                  " due to job rescheduling"
+        print
```

Comment:
Are the new prints in this file meant to reside in the code, or are they the result of debugging work?

Reply:
Print without parens is a python 3 incompatibility. Parens also eliminate the need to escape line breaks.